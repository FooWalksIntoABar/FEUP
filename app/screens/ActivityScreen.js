import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { setActivityStatus, nextActivity } from '../actions/gameActions'
import { addStars } from '../actions/childActions'
import { Image, Text, View, StatusBar, BackHandler } from 'react-native'

import { ProgressBar } from '../components/Activity/ProgressBar'
import { ProgressClock } from '../components/Activity/ProgressClock'
import { CompleteButton } from '../components/Activity/CompleteButton'
import { PauseButton } from '../components/Activity/PauseButton'
import { CancelButton } from '../components/Activity/CancelButton'
import { RewardsModal } from '../components/RewardsModal/RewardsModal'
import Images from '../assets/images/images'

import styles from '../styles/Activity.style'

class ActivityScreen extends Component {
  constructor (props) {
    super(props)

    this.state = {
      elapsedTime: 0,
      progressType: '',
      isPhoto: this.props.activity.photo !== undefined,
      updateRate: 100, // ms
      isPaused: false,
      isCompleted: false,
      isCompletable: false
    }

    this.pauseActivity = this.pauseActivity.bind(this)
    this.cancelActivity = this.cancelActivity.bind(this)
    this.completeActivity = this.completeActivity.bind(this)
    this.resumeActivity = this.resumeActivity.bind(this)
    this.nextActivity = this.nextActivity.bind(this)
    this.backToMenu = this.backToMenu.bind(this)
  }

  componentDidMount () {
    BackHandler.addEventListener('hardwareBackPres', this.cancelActivity)
    this.interval = setInterval(() => {
      if (this.state.isPaused) return
      if (this.state.elapsedTime >= this.props.activity.time.max) this.completeActivity()
      if (this.state.elapsedTime >= this.props.activity.time.min) {
        this.state.isCompletable = true
      }

      this.setState(() => {
        return { elapsedTime: this.state.elapsedTime + this.state.updateRate / 1000 }
      })
    }, this.state.updateRate)
  }

  componentWillUnmount () {
    BackHandler.removeEventListener('hardwareBackPress', this.cancelActivity)
    clearInterval(this.interval)
  }

  pauseActivity () {
    this.setState(() => ({ isPaused: true }))
  }

  cancelActivity () {
    this.props.navigation.popToTop()
    return true // To be called by andrdid back button
  }

  completeActivity () {
    clearInterval(this.interval)
    const status = {
      completed: true,
      reward: this.state.elapsedTime > this.props.activity.time.max ? 0
        : this.state.elapsedTime < this.props.activity.time.goal ? 3
          : this.state.elapsedTime < this.props.activity.time.goal + (this.props.activity.time.max - this.props.activity.time.goal) / 2 ? 2 : 1,
      time: parseInt(this.state.elapsedTime)
    }
    this.props.setActivityStatus(this.props.activity, status)
    this.setState(() => { return { isCompleted: true } })
  }

  nextActivity () {
    if (this.props.activity.status) this.props.addStars(this.props.activity.status.reward)
    this.props.nextActivity()
    this.props.navigation.replace('Activity')
  }

  backToMenu () {
    if (this.props.activity.status) this.props.addStars(this.props.activity.status.reward)
    this.props.navigation.popToTop()
  }

  resumeActivity () {
    this.setState(() => ({ isPaused: false }))
  }

  returnURIorImage = () => {
    // TODO: This checks whether the photo attribute is type URI and should probably just eventually be totally changed to URI.
    if (this.props.activity.photo !== undefined && this.props.activity.photo.includes('file://')) {
      return { uri: this.props.activity.photo }
    } else {
      return Images[this.state.isPhoto ? this.props.activity.photo : this.props.activity.image]
    }
  }

  render () {
    return (
      <View style={[{ backgroundColor: this.props.activity.color }, styles.activityScreen]} >
        <StatusBar hidden />
        <Image
          style={this.state.isPhoto ? styles.photo : styles.image}
          resizeMode={this.state.isPhoto ? 'cover' : 'center'}
          source={this.returnURIorImage()} />
        <View style={styles.titleContainer}>
          <Text style={this.state.isPhoto ? styles.photoTitle : styles.title}>{this.props.activity.title}</Text>
        </View>
        {this.props.progressType === 'clock' && !this.state.isCompleted && <ProgressClock showTimer={this.props.showTimer} elapsedTime={this.state.elapsedTime} activityTimes={this.props.activity.time} isPaused={this.state.isPaused} />}
        {!this.state.isCompleted && <View style={styles.buttonContainer}>
          {this.props.progressType === 'bar' && <ProgressBar showTimer={this.props.showTimer} elapsedTime={this.state.elapsedTime} activityTimes={this.props.activity.time} isPaused={this.state.isPaused} />}
          <CancelButton style={styles.smallButton} cancelActivity={this.cancelActivity} />
          <PauseButton style={styles.smallButton} pauseActivity={this.pauseActivity} resumeActivity={this.resumeActivity} isPaused={this.state.isPaused} />
          <CompleteButton style={styles.largeButton} isCompletable={this.state.isCompletable} completeActivity={this.completeActivity} />
        </View>}
        <RewardsModal
          currentActivity={this.props.currentActivity}
          activities={this.props.activities}
          nextPress={this.nextActivity}
          backPress={this.backToMenu} />
      </View>
    )
  }
}

export default connect(
  state => ({
    progressType: state.settings.activityProgressType,
    showTimer: state.settings.activityShowTimer,
    activity: state.game.routines[state.game.currentRoutine].activities[state.game.currentActivity],
    activities: state.game.routines[state.game.currentRoutine].activities,
    currentActivity: state.game.currentActivity
  }),
  dispatch => ({
    setActivityStatus: (activity, status) => dispatch(setActivityStatus(activity, status)),
    nextActivity: () => dispatch(nextActivity()),
    addStars: stars => dispatch(addStars(stars))
  })
)(ActivityScreen)

ActivityScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  progressType: PropTypes.string.isRequired,
  showTimer: PropTypes.bool.isRequired,
  currentActivity: PropTypes.number.isRequired,
  activities: PropTypes.array.isRequired,
  activity: PropTypes.object.isRequired,
  setActivityStatus: PropTypes.func.isRequired,
  nextActivity: PropTypes.func.isRequired,
  addStars: PropTypes.func.isRequired
}
