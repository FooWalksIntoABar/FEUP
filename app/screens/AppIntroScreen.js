import React, { Component } from 'react'
import AppIntroSlider from 'react-native-app-intro-slider'
import PropTypes from 'prop-types'
import Images from '../assets/images/images'

const slides = [
  {
    key: '1',
    title: 'Texto explicativo #1',
    text: 'A ser preenchido pelas nossas fantásticas product owners.\nO número de páginas e imagem é ajustável consoante necessário.',
    image: Images.socks,
    imageStyle: { width: 240, height: 240 },
    backgroundColor: '#59b2ab'
  },
  {
    key: '2',
    title: 'Texto explicativo #2',
    text: 'A ser preenchido pelas nossas fantásticas product owners.\nO número de páginas e imagem é ajustável consoante necessário.',
    image: Images.socks,
    imageStyle: { width: 240, height: 240 },
    backgroundColor: '#febe29'
  },
  {
    key: '3',
    title: 'Texto explicativo #3',
    text: 'A ser preenchido pelas nossas fantásticas product owners.\nO número de páginas e imagem é ajustável consoante necessário.',
    image: Images.socks,
    imageStyle: { width: 240, height: 240 },
    backgroundColor: '#22bcb5'
  }
]

export default class AppIntroScreen extends Component {
  _onDone = () => {
    this.props.navigation.goBack()
  }

  render () {
    return (
      <AppIntroSlider slides={slides} onDone={this._onDone} />
    )
  }
}

AppIntroScreen.propTypes = {
  navigation: PropTypes.object.isRequired
}
