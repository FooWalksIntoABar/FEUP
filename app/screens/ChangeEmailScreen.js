import React from 'react'
import { View, Text, TextInput, TouchableOpacity } from 'react-native'

import PropTypes from 'prop-types'

import styles from '../styles/ParentStyles/ChangeEmailScreen.style'

export default class ChangeEmailScreen extends React.Component {
  static navigationOptions = {
    headerMode: 'screen',
    navigationOptions: {
      headerTitle: 'Repor E-mail'
    }
  };
  render () {
    return (
      <View style={styles.generalLayout}>
        <View style={styles.container}>
          <Text>E-mail Atual</Text>
          <Text>{this.props.navigation.state.params.email}</Text>
        </View>
        <View style={styles.container}>
          <Text>Novo E-mail</Text>
          <TextInput
            keyboardType='email-address'
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View style={styles.container}>
          <Text>Password</Text>
          <Text>Insira a sua password para confirmar a alteração de e-mail.</Text>
          <TextInput
            secureTextEntry
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View >
          <TouchableOpacity style={styles.confirmBtn} onPress={() => { this.props.navigation.state.params.setEmail('hahahha') }}>
            <Text>Confirmar</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}

ChangeEmailScreen.propTypes = {
  navigation: PropTypes.object.isRequired
}
