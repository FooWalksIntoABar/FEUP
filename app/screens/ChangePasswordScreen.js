import React from 'react'
import { View, Text, TextInput, TouchableOpacity } from 'react-native'

import styles from '../styles/ParentStyles/ChangePasswordScreen.style'

export default class ChangePasswordScreen extends React.Component {
  static navigationOptions = {
    headerMode: 'screen',
    navigationOptions: {
      headerTitle: 'Repor E-mail'
    }
  };
  render () {
    return (
      <View style={styles.generalLayout}>
        <View style={styles.container}>
          <Text>Insira a sua antiga password.</Text>
          <TextInput
            secureTextEntry
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View style={styles.container}>
          <Text>Insira a sua nova password.</Text>
          <TextInput
            secureTextEntry
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View style={styles.container}>
          <Text>Repita a sua nova password.</Text>
          <TextInput
            secureTextEntry
            placeholder='Insira o seu novo e-mail'
            style={styles.inputContainer}
          />
        </View>
        <View>
          <TouchableOpacity onPress={() => { console.log('changed password') }}>
            <Text>Alterar Password</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}
