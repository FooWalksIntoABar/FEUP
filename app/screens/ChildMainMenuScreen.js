import React, { Component } from 'react'
import { View, StatusBar, Image, TouchableOpacity } from 'react-native'
import { ScreenOrientation } from 'expo'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { ChildExperienceBar } from '../components/MainMenu/ChildExperienceBar'

import Images from '../assets/images/images'

import styles from '../styles/ChildMainMenuScreen.style'

class ChildMainMenuScreen extends Component {
  constructor (props) {
    super(props)
    this.openShop = this.openShop.bind(this)
    this.closeShop = this.closeShop.bind(this)
  }

  state = {
    isShopVisible: false
  }

  componentDidMount () {
    ScreenOrientation.allow(ScreenOrientation.Orientation.LANDSCAPE)
  }

  componentWillUnmount () {
    ScreenOrientation.allow(ScreenOrientation.Orientation.PORTRAIT)
  }

  openShop () {
    this.setState({ isShopVisible: true })
  }

  closeShop () {
    this.setState({ isShopVisible: false })
  }

  render () {
    return (
      <View style={styles.mainMenuContainer}>
        <StatusBar hidden />
        <View style={styles.experienceBarContainer}>
          <ChildExperienceBar
            progress={(this.props.xp - this.props.level * 100) / 100}
            level={this.props.level} />
        </View>
        <View style={styles.buttonContainer}>
          <TouchableOpacity
            style={styles.profileButton}
            onPress={() => { this.props.navigation.navigate('ShopScreen') }}
            activeOpacity={0.9} >
            <Image style={styles.profileImage} resizeMode={'contain'} source={Images.avatar} />
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.button}
            onPress={() => { this.props.navigation.navigate('ChooseRoutineScreen') }}
            activeOpacity={0.9}>
            <Image style={styles.buttonImage} source={Images.ui.play} />
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}

export default connect(
  state => ({
    level: state.child.level,
    xp: state.child.xp
  })
)(ChildMainMenuScreen)

ChildMainMenuScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  level: PropTypes.number.isRequired,
  xp: PropTypes.number.isRequired
}
