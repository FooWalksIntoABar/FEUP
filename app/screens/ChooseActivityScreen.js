import React from 'react'
import { StatusBar, View } from 'react-native'
import { connect } from 'react-redux'
import { setCurrentActivity } from '../actions/gameActions'
import PropTypes from 'prop-types'

import CardCarousel from '../components/Carousel/CardCarousel'
import { backgroundColor } from '../styles/Colors'

export class ChooseActivityScreen extends React.Component {
  constructor (props) {
    super(props)
    this.onPress = this.onPress.bind(this)
  }

  onPress = (activity) => {
    this.props.setCurrentActivity(activity)
    this.props.navigation.navigate('Activity')
  }

  render () {
    return (
      <View style={{ backgroundColor: backgroundColor }}>
        <StatusBar hidden />
        <CardCarousel
          data={this.props.activities.filter(activity => activity.status === undefined)}
          onPress={this.onPress} />
      </View>
    )
  }
}

export default connect(
  state => ({
    activities: state.game.routines[state.game.currentRoutine].activities
  }),
  dispatch => ({
    setCurrentActivity: routine => dispatch(setCurrentActivity(routine))
  })
)(ChooseActivityScreen)

ChooseActivityScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  activities: PropTypes.array.isRequired,
  setCurrentActivity: PropTypes.func.isRequired
}
