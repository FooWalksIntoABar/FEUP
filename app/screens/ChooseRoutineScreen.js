import React from 'react'
import { StatusBar, View, Text } from 'react-native'
import { connect } from 'react-redux'
import { setCurrentRoutine, nextActivity } from '../actions/gameActions'
import PropTypes from 'prop-types'

import CardCarousel from '../components/Carousel/CardCarousel'
import styles from '../styles/CardCarousel.style'

class ChooseRoutineScreen extends React.Component {
  constructor (props) {
    super(props)
    this.chooseActivities = this.chooseActivities.bind(this)
    this.startRoutine = this.startRoutine.bind(this)
  }

  static navigationOptions = {
    header: null
  }

  chooseActivities = (routine) => {
    this.props.setCurrentRoutine(routine)
    this.props.navigation.navigate('ChooseActivityScreen')
  }

  startRoutine = (routine) => {
    this.props.setCurrentRoutine(routine)
    this.props.nextActivity()
    this.props.navigation.navigate('Activity')
  }

  render () {
    if (!this.props.routines.every(routine => routine.activities.every(activity => activity.status !== undefined))) {
      return (
        <View style={styles.screenBackground}>
          <StatusBar hidden />
          <CardCarousel
            data={this.props.routines.filter(routine => !routine.activities.every(activity => activity.status !== undefined))}
            onPress={this.props.routinePlayType === 'auto' ? this.startRoutine : this.chooseActivities}
            onButtonPress={this.props.routinePlayType === 'choose' ? this.startRoutine : undefined}
            isRoutine
          />
        </View>
      )
    } else {
      return (<View style={[styles.screenBackground, { justifyContent: 'center', alignItems: 'center' }]}>
        <Text style={styles.tooltip}>Sem rotinas para jogar.{'\n'}Volta amanh√£!</Text>
      </View>)
    }
  }
}

export default connect(
  state => ({
    routines: state.game.routines,
    routinePlayType: state.settings.routinePlayType
  }),
  dispatch => ({
    setCurrentRoutine: routine => dispatch(setCurrentRoutine(routine)),
    nextActivity: () => dispatch(nextActivity())
  })
)(ChooseRoutineScreen)

ChooseRoutineScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  routinePlayType: PropTypes.string.isRequired,
  routines: PropTypes.array.isRequired,
  setCurrentRoutine: PropTypes.func.isRequired,
  nextActivity: PropTypes.func.isRequired
}
