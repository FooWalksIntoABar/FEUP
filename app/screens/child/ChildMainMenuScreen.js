import React, { Component } from 'react'
import { View, StatusBar, Image, TouchableOpacity } from 'react-native'
import { ScreenOrientation } from 'expo'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { toggleLevelUpModal } from '../../actions/childActions'
import { addRoutines } from '../../actions/gameActions'
import { ChildExperienceBar } from '../../components/ChildMainMenu/ChildExperienceBar'
import { LevelUpModal } from '../../components/ChildMainMenu/LevelUpModal'
import { Avatar } from '../../components/ChildMainMenu/Avatar'
import EnvVars from '../../constants/EnviromentVars'

import Images from '../../assets/images/images'

import styles from '../../styles/ChildMainMenuScreen.style'

export class ChildMainMenuScreen extends Component {
  constructor (props) {
    super(props)

    this.state = {
      isShopVisible: false,
      showModal: false,
      reward: { name: '', photo: undefined }
    }

    this.openShop = this.openShop.bind(this)
    this.closeShop = this.closeShop.bind(this)
    this.onCloseModal = this.onCloseModal.bind(this)
    this.showRewardModal = this.showRewardModal.bind(this)
  }

  componentDidMount () {
    ScreenOrientation.allow(ScreenOrientation.Orientation.LANDSCAPE)
  }

  componentWillUnmount () {
    ScreenOrientation.allow(ScreenOrientation.Orientation.PORTRAIT)
  }

  componentWillMount () {
    let url = `${EnvVars.apiUrl}routine_manager/get-daily-routine?selectedChildID=${this.props.childID}`
    fetch(url)
      .then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.status === '200') {
          this.props.addRoutines(JSON.parse(responseJson.response))
        } else {

        }
        return responseJson
      })
      .catch((error) => {
        console.error(error)
      })
    this.fetchChildReward()
  }

  fetchChildReward = () => {
    let url = `${EnvVars.apiUrl}routine_manager/get-reward?selectedChildID=${this.props.childID}`
    fetch(url)
      .then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.status === '200') {
          let reward = JSON.parse(responseJson.response).map(reward => ({ title: reward.name, ...reward }))[0]
          if (reward) {
            this.setState({ reward, showModal: this.props.showLevelUpModal })
          } else {
            this.setState({ reward: { name: '', photo: undefined } })
          }
        } else {

        }
        return responseJson
      })
      .catch((error) => {
        console.error(error)
      })
  }

  deleteReward (id) {
    fetch(EnvVars.apiUrl + 'routine_manager/remove-reward/', {
      method: 'DELETE',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        rewardID: id,
        childID: this.props.childID
      })
    }).then((response) => response.json())
      .then((responseJson) => {
        if (responseJson.status === '200') {
          console.log('trofeu apagada')
          this.fetchChildReward()
        } else {
          console.log('trofeu nao apagado')
        }
        return responseJson
      })
      .catch((error) => {
        console.error(error)
      })
  }

  openShop () {
    this.setState({ isShopVisible: true })
  }

  closeShop () {
    this.setState({ isShopVisible: false })
  }

  onCloseModal () {
    this.setState({ showModal: false })
    if (this.props.showLevelUpModal) {
      this.props.toggleLevelUpModal()
      this.deleteReward(this.state.reward.id)
    }
  }

  showRewardModal () {
    this.setState({ showModal: true })
  }

  render () {
    return (
      <View style={styles.mainMenuContainer}>
        <StatusBar hidden />
        <View style={styles.experienceBarContainer}>
          <ChildExperienceBar
            progress={(this.props.xp - this.props.level * 100) / 100}
            level={this.props.level}
            onPress={this.showRewardModal}
            photo={this.state.reward.photo} />
        </View>
        <View style={styles.buttonContainer}>
          <TouchableOpacity
            style={styles.profileButton}
            onPress={() => { this.props.navigation.navigate('ShopScreen') }}
            activeOpacity={0.9} >
            <Avatar button equiped={this.props.itemsEquiped} gender={this.props.gender} />
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.button}
            onPress={() => { this.props.navigation.navigate('ChooseRoutineScreen') }}
            activeOpacity={0.9}>
            <Image style={styles.buttonImage} source={Images.ui.play} />
          </TouchableOpacity>
        </View>
        {this.state.showModal && <LevelUpModal
          level={this.props.level}
          xp={this.props.xp}
          isReward={this.props.showLevelUpModal}
          show={this.state.showModal}
          onClosed={this.onCloseModal}
          playSounds={this.props.playSounds}
          reward={this.state.reward} />}
      </View>
    )
  }
}

export default connect(
  /* istanbul ignore next */
  state => ({
    childID: state.child.id,
    level: state.child.level,
    xp: state.child.xp,
    gender: state.child.gender,
    itemsEquiped: state.child.itemsEquiped,
    showLevelUpModal: state.child.showLevelUpModal,
    playSounds: state.settings.playSounds
  }),
  /* istanbul ignore next */
  dispatch => ({
    toggleLevelUpModal: () => dispatch(toggleLevelUpModal()),
    addRoutines: routines => dispatch(addRoutines(routines))
  })
)(ChildMainMenuScreen)

ChildMainMenuScreen.propTypes = {
  navigation: PropTypes.object.isRequired,
  level: PropTypes.number.isRequired,
  xp: PropTypes.number.isRequired,
  showLevelUpModal: PropTypes.bool.isRequired,
  toggleLevelUpModal: PropTypes.func.isRequired,
  playSounds: PropTypes.bool.isRequired,
  gender: PropTypes.string.isRequired,
  itemsEquiped: PropTypes.array.isRequired,
  childID: PropTypes.number.isRequired,
  addRoutines: PropTypes.func.isRequired
}
