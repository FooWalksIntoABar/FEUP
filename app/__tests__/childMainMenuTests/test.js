import React from 'react'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { TouchableOpacity } from 'react-native'

import { ChildMainMenuScreen } from '../../screens/ChildMainMenuScreen'

configure({ adapter: new Adapter() })

describe('ChildMainMenuScreen snapshot', () => {
  it('renders ChildMainMenuScreen correctly', () => {
    const wrapper = shallow(<ChildMainMenuScreen
      navigation={{ navigate: jest.fn() }}
      xp={150}
      level={1}
      showLevelUpModal={false}
      toggleLevelUpModal={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find(TouchableOpacity).at(0).props().onPress()
    wrapper.find(TouchableOpacity).at(1).props().onPress()
    wrapper.instance().openShop()
    wrapper.instance().closeShop()
    wrapper.unmount()
  })
})
