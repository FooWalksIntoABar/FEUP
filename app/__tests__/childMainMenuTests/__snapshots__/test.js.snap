// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChildMainMenuScreen snapshot renders ChildMainMenuScreen correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff7ae",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "flex": 0,
      }
    }
  >
    <View
      style={
        Object {
          "height": 213.44,
          "justifyContent": "flex-end",
          "paddingLeft": 106.72,
          "paddingRight": 160.07999999999998,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "gray",
              "borderRadius": 53.36,
              "elevation": 6,
              "height": 106.72,
              "overflow": "hidden",
              "position": "relative",
              "width": "100%",
            },
          ]
        }
      >
        <View
          animated={true}
          animationConfig={
            Object {
              "bounciness": 0,
            }
          }
          animationType="spring"
          indeterminate={false}
          onLayout={[Function]}
          progress={0.5}
          style={
            Array [
              Object {
                "backgroundColor": undefined,
                "borderColor": "#700548",
                "borderRadius": 0,
                "borderWidth": 0,
                "overflow": "hidden",
                "width": null,
              },
              undefined,
            ]
          }
          useNativeDriver={false}
        >
          <View
            style={
              Object {
                "backgroundColor": "#700548",
                "height": 106.72,
                "transform": Array [
                  Object {
                    "translateX": -0,
                  },
                  Object {
                    "translateX": -0,
                  },
                  Object {
                    "scaleX": 0.50005,
                  },
                ],
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "left": "25%",
              },
              Object {
                "borderRightColor": "#000",
                "borderRightWidth": 2,
                "height": 106.72,
                "opacity": 0.5,
                "position": "absolute",
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "left": "50%",
              },
              Object {
                "borderRightColor": "#000",
                "borderRightWidth": 2,
                "height": 106.72,
                "opacity": 0.5,
                "position": "absolute",
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "left": "75%",
              },
              Object {
                "borderRightColor": "#000",
                "borderRightWidth": 2,
                "height": 106.72,
                "opacity": 0.5,
                "position": "absolute",
              },
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 80.03999999999999,
            "bottom": 26.68,
            "elevation": 7,
            "height": 160.07999999999998,
            "overflow": "hidden",
            "position": "absolute",
            "right": 80.03999999999999,
            "width": 160.07999999999998,
          }
        }
      >
        <Image
          resizeMode="cover"
          source={1}
          style={
            Object {
              "height": 160.07999999999998,
              "width": 160.07999999999998,
            }
          }
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#000",
            "fontFamily": "LinotteBold",
            "fontSize": 18,
            "height": 53.36,
            "marginLeft": 41.57333333333333,
            "textShadowColor": "rgba(0, 0, 0, 0.2)",
            "textShadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "textShadowRadius": 3,
          }
        }
      >
        NÃ­vel 
        1
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#49c6e5",
          "borderRadius": 187.5,
          "elevation": 10,
          "height": 375,
          "justifyContent": "center",
          "opacity": 1,
          "overflow": "hidden",
          "position": "relative",
          "width": 375,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": 562.5,
            "position": "absolute",
            "top": 0,
            "width": 562.5,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#16db65",
          "borderRadius": 187.5,
          "elevation": 10,
          "height": 375,
          "justifyContent": "center",
          "opacity": 1,
          "overflow": "hidden",
          "position": "relative",
          "width": 375,
        }
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": "40%",
            "marginLeft": "7%",
            "tintColor": "#fff",
            "width": "40%",
          }
        }
      />
    </View>
  </View>
</View>
`;
